platforms/android/app/src/main/java/com/maq/xprize/cci/hindi/SplashScreenActivity.java

```diff
@@ -113,28 +137,54 @@ public void toCallApplication() {
     }
 
     public void unzipFile() {
+        int totalSize = getTotalSize();
+        sharedPref = getSharedPreferences("ExpansionFile", MODE_PRIVATE);
+        mainFileVersion = sharedPref.getInt(getString(R.string.mainFileVersion), 0);
+        patchFileVersion = sharedPref.getInt(getString(R.string.patchFileVersion), 0);
         try {
-            String filePath = getExpansionFilePath();
-            File file = new File(filePath);
-            ZipFile zipFile = new ZipFile(file);
-            Zip _zip = new Zip(zipFile, this);
-            String unzipFilePath = getUnzippedExpansionFilePath();
-            File packageNameDir = new File(unzipFilePath);
-            if (packageNameDir.exists()) {
-                DownloadExpansionFile.deleteDir(packageNameDir);
+            for (DownloadExpansionFile.XAPKFile xf : xAPKs) {
+                if (xf.mIsMain && xf.mFileVersion != mainFileVersion || !xf.mIsMain && xf.mFileVersion != patchFileVersion) {
+                    obbFilePath = getObbFilePath(xf.mIsMain, xf.mFileVersion);
+                    obbFile = new File(obbFilePath);
+                    obbZipFile = new ZipFile(obbFile);
+                    zipFileHandler = new Zip(obbZipFile, this);
+                    dataFilePath = getDataFilePath();
+                    packageDir = new File(dataFilePath);
+                    if (xf.mIsMain && packageDir.exists()) {
+                        DownloadExpansionFile.deleteDir(packageDir);
+                        packageDir.mkdir();
+                    }
+                    zipFileHandler.unzip(dataFilePath, totalSize, xf.mIsMain, xf.mFileVersion);
+                    zipFileHandler.close();
+                }
             }
-            packageNameDir.mkdir();
-            _zip.unzip(unzipFilePath);
-            _zip.close();
             toCallApplication();
+        } catch (IOException e) {
+            System.out.println(e);
+        }
+    }
+
+    public int getTotalSize() {
+        int totalSize = 0;
+        try {
+            for (DownloadExpansionFile.XAPKFile xf : xAPKs) {
+                if (!xf.mIsMain && (xf.mFileVersion != patchFileVersion) || xf.mIsMain && (xf.mFileVersion != mainFileVersion)) {
+                    obbFilePath = getObbFilePath(xf.mIsMain, xf.mFileVersion);
+                    obbFile = new File(obbFilePath);
+                    obbZipFile = new ZipFile(obbFile);
+                    totalSize += obbZipFile.size();
+
+                }
+            }
         } catch (IOException ie) {
-            System.err.println(ie);
+            System.out.println(ie);
         }
+        return totalSize;
     }
 
-    public String getExpansionFilePath() {
+    public String getObbFilePath(boolean isMain, int fileVersion) {
         return Environment.getExternalStorageDirectory().toString() + "/Android/obb/" + Helpers.getPackageName(this) + File.separator +

```

Comment:
Use `getObbDir()` instead of `Environment.getExternalStorageDirectory().toString() + "/Android/obb/" + Helpers.getPackageName(this)`

Reply:
Updated